cfme/tests/intelligence/reports/test_validate_chargeback_report.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+
+import cfme.intelligence.chargeback as cb
+import fauxfactory
+import pytest
+import time
+
+from cfme.intelligence.reports.reports import CustomReport
+from utils.providers import setup_a_provider
+from utils.log import logger
+
+
+@pytest.fixture(scope="module")
+def setup_first_provider():
+    setup_a_provider(validate=True, check_existing=True)
+
+
+@pytest.fixture(scope="module")
+def sleep():
+    # Chargeback reporting is not done on real-time values and hence we have to wait for atleast an
+    # hour for the data rollups to happen.Once the hourly rollups records become available in the
+    # DB, Chargeback reports can be generated.
+    #
+    # In these tests,we collect C&U data for 2 hours.
+    time.sleep(7200)
+
+
+@pytest.fixture(scope="module")
+def assign_compute_default_rate(provider):
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            "Enterprise": "Default"
+        })
+    enterprise.computeassign()
+
+
+@pytest.fixture(scope="module")
+def assign_compute_custom_rate(provider):
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            "Enterprise": "Default"
+        })
+    enterprise.computeassign()
+
+
+@pytest.fixture(scope="module")
+def assign_storage_default_rate(provider):
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            "Enterprise": "Default"
+        })
+
+    enterprise.storageassign()
+
+
+@pytest.fixture(scope="module")
+def assign_storage_custom_rate(provider):
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            "Enterprise": "Default"
+        })
+
+    enterprise.storageassign()
+
+
+class cost:
```

Comment:
Two things here, 
#1 Why use a class for this? Wouldn't a dict do?
#2 Is it supposed to be instantiated and used at the module level?